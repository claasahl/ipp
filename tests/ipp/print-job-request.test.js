const fs = require("fs");

const encode = require("../../build/ipp/encode").default;

test("adds 1 + 2 to equal 3", () => {
  const data = encode({
    type: "IppRequest",
    versionNumber: { major: 0x1, minor: 0x1 },
    operationId: 0x2,
    requestId: 0x1,
    attributeGroup: [
      {
        type: "AttributeGroup",
        beginAttributeGroupTag: 0x1,
        attribute: [
          {
            type: "Attribute",
            attributeWithOneValue: {
              type: "AttributeWithOneValue",
              valueTag: 0x47,
              nameLength: 0x12,
              name: "attributes-charset",
              valueLength: 0x5,
              value: "utf-8"
            },
            additionalValue: []
          },
          {
            type: "Attribute",
            attributeWithOneValue: {
              type: "AttributeWithOneValue",
              valueTag: 0x48,
              nameLength: 0x1b,
              name: "attributes-natural-language",
              valueLength: 0x5,
              value: "en-us"
            },
            additionalValue: []
          },
          {
            type: "Attribute",
            attributeWithOneValue: {
              type: "AttributeWithOneValue",
              valueTag: 0x45,
              nameLength: 0xb,
              name: "printer-uri",
              valueLength: 0x2c,
              value: "ipp://printer.example.com/ipp/print/pinetree"
            },
            additionalValue: []
          },
          {
            type: "Attribute",
            attributeWithOneValue: {
              type: "AttributeWithOneValue",
              valueTag: 0x42,
              nameLength: 0x8,
              name: "job-name",
              valueLength: 0x6,
              value: "foobar"
            },
            additionalValue: [
              {
                type: ""
              }
            ]
          }
        ]
      }
    ],
    endOfAttributesTag: 3,
    data: Buffer.from("%!PDF...")
  });
  expect(data.toString("hex")).toMatchSnapshot();
});
