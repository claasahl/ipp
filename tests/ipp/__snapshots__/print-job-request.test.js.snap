// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Print-Job Request 1`] = `"010100020000000101470012617474726962757465732d6368617273657400057574662d3848001b617474726962757465732d6e61747572616c2d6c616e67756167650005656e2d757345000b7072696e7465722d757269002c6970703a2f2f7072696e7465722e6578616d706c652e636f6d2f6970702f7072696e742f70696e65747265654200086a6f622d6e616d650006666f6f6261722200166970702d6174747269627574652d666964656c69747900010102210006636f706965730004000000144400057369646573001374776f2d73696465642d6c6f6e672d656467650325215044462e2e2e"`;
